Flw v3 (#39)
ashinzekene/angular-rave: README.md

You can checkout the demo [here](https://ashinzekene.github.io/angular-rave)

### 1. Install the module

>   Run in the your angular project

Author: ```suggestion
  Run this in the root of your angular project
```

  @NgModlule({
    imports: [
      AngularRaveModule.forRoot('FLWPUBK-XXXXXXXXXXXXXXXXXXX'),
    ]
  })
  ```

>   where `FLWPUBK-XXXXXXXXXXXXXXXXXXX` is your public which can be found on the flutterwave dashboard

Author: ```suggestion
  where `FLWPUBK-XXXXXXXXXXXXXXXXXXX` is your public key which can be found on the flutterwave dashboard
```

- The `angular-rave` component:
```html
<angular-rave
  [customer]="{ email: 'user@example.com', phonenumber: '0809808800900' }"
  [amount]="500000"

>   [customizations]="{ title: 'Bill Payment'}"

Author: ```suggestion
  [customizations]="{ title: 'Bill Payment' }"
```

  [tx_ref]="'USR1295950'"
  (callback)="paymentSuccess($event)"
  (close)="paymentFailure()"

>   (close)="paymentInit()"

Author: ```suggestion
  (init)="paymentInit()"
```

```html
<button
  angular-rave
  [customer]="{ email: 'user@example.com', phonenumber: '0809808800900' }"
  [amount]="500000"
  [customizations]="{ title: 'Bill Payment'}"
  [tx_ref]="'USR1295950'"
  (callback)="paymentSuccess($event)"
  (close)="paymentFailure()"

>   (close)="paymentInit()"

Author: ```suggestion
  (init)="paymentInit()"
```

```html
<button
  angular-rave
  [raveOptions]="raveOptions"
  (init)="paymentInit()"
  (callback)="paymentSuccess($event)"
  (onclose)="paymentFailure()"
>
  Pay with Rave Directive
</button>

```
And then you can import the `RaveOptions` class for help in typing
```ts

> import { RaveOptions, PaymentData } from 'angular-rave';

Author: ```suggestion
import { RaveOptions } from 'angular-rave';
```

| redirect_url            | string                    | false             | -               | URL to redirect to when a transaction is completed. This is useful for 3DSecure payments so we can redirect your customer back to a custom page you want to show them.
| customer                | RaveCustomer              | true              | -               | This is an object that can contains your customer details: e.g: `{ 'email': 'example@example.com', 'phonenumber': '08012345678', 'name': 'Takeshi Kovacs'}`
| meta                    | `{[key: string]: any}`    | false             | -               | This is an object that helps you include additional payment information to your request `{'consumer_id': 23,'consumer_mac': '92a3-912ba-1192a'}`
| customizations          | `RaveCustomization`       | true              | -               | This is an object that contains title, logo, and description you want to display on the modal e.g `{'title': 'Pied Piper Payments', 'description': 'Middleout isn't free. Pay the price', 'logo': 'https://assets.piedpiper.com/logo.png'}`

> | init                    | function(res)             | false             | -               | A function to be called when payment is about to begin

Author: ```suggestion
| init                    | `() => void`           | false             | -               | A function to be called when payment is about to begin
```

| onclose                 | function()                | false             | -               | A function to be called when the pay modal is closed before a transaction is completed.

> | callback                | function(res)             | true              | -               | A function to be called on successful card charge. User’s can always be redirected to a successful or failed page supplied by the merchant here based on response.

Author: ```suggestion
| callback                | (res: PaymentData) => void     | true              | -               | A function to be called on successful card charge. Users can always be redirected to a successful or failed page supplied by the merchant here based on response.
```
